addons/mail/static/src/messaging/messaging_env.js

```diff
@@ -43,12 +41,8 @@ function addMessagingToEnv(env) {
     /**
      * Messaging entities.
      */
-    let messagingCreatedPromise;
-    if (env.generateEntitiesImmediately) {
-        _addMessagingEntities(env);
-        messagingCreatedPromise = Promise.resolve();
-    } else {
-        messagingCreatedPromise = new Promise(resolve => {
+    if (!env.generateEntitiesImmediately) {
+        await new Promise(resolve => {
```

Comment:
is this promise really necessary?

Reply:
We have an event handler that is async by definition and we want to wait on it so yes? What do you suggest?

Reply:
oops misread, you can ignore this comment